DEVELOPMENT.md

```diff
@@ -0,0 +1,71 @@
+**This is not an officially supported Google product.**
+
+[![Build Status](https://travis-ci.org/googleinterns/knative-continuous-delivery.svg?branch=main)](https://travis-ci.org/googleinterns/knative-continuous-delivery)
+
+# Knative Continuous Delivery
+
+## Development
+
+### Dependencies
+
+1. [go](https://golang.org/doc/install)
+
+### Setup environment
+
+Put the following in a `./bashrc` or `./bashprofile`
+
+```sh
+export GOPATH="$HOME/go"
+export PATH="${PATH}:${GOPATH}/bin"
+```
+
+### Clone to your machine
+
+1. [Fork this repo](https://help.github.com/articles/fork-a-repo/) to your account
+2. Clone to your machine
+
+```sh
+mkdir -p ${GOPATH}/src/github.com/googleinterns
+cd ${GOPATH}/src/github.com/googleinterns
+git clone git@github.com:${YOUR_GITHUB_USERNAME}/knative-continuous-delivery.git
+cd knative-continuous-delivery
+git remote add upstream https://github.com/googleinterns/knative-continuous-delivery.git
+git remote set-url --push upstream no_push
+```
+
+`Upstream` will allow for you to [sync your fork](https://help.github.com/articles/syncing-a-fork/)
+
+### Building
+
+`go build ./...`
+
+### Running unit tests
+
+`go test ./...`
+
+### Updating dependencies
+
+`./hack/update-deps.sh`
```

Comment:
do we run ./update-codegen.sh

Reply:
I don't think we need that, because all the auto generated codes have been submitted to this repo

Reply:
I think @vagababov is saying to replace this with `./hack/update-codegen.sh` because this also updates the deps: https://github.com/googleinterns/knative-continuous-delivery/blob/97d8aa7d6640e08e8efd6b9684089a5fedcc0dcc/hack/update-codegen.sh#L59

Reply:
I see what you mean. Fixed